examples/text-classification/run_pl_glue.py

```diff
@@ -1,115 +1,205 @@
-import argparse
 import glob
 import logging
 import os
-import time
+from argparse import ArgumentParser, Namespace
+from typing import List, Union
 
+import nlp
 import numpy as np
+import pandas as pd
+import pyarrow
 import torch
-from torch.utils.data import DataLoader, TensorDataset
+from pytorch_lightning import Trainer
+from torch.utils.data import DataLoader
 
-from lightning_base import BaseTransformer, add_generic_args, generic_train
-from transformers import glue_compute_metrics as compute_metrics
-from transformers import glue_convert_examples_to_features as convert_examples_to_features
-from transformers import glue_output_modes
-from transformers import glue_processors as processors
-from transformers import glue_tasks_num_labels
+from lightning_base import BaseTransformer, LoggingCallback, set_seed
+from transformers.data import glue_tasks_num_labels
 
 
 logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+# Glue tasks in nlp library have names without dashes, so we remove them
+glue_tasks_num_labels = {k.replace("-", ""): v for k, v in glue_tasks_num_labels.items()}
 
 
 class GLUETransformer(BaseTransformer):
 
     mode = "sequence-classification"
 
-    def __init__(self, hparams):
-        hparams.glue_output_mode = glue_output_modes[hparams.task]
+    def __init__(self, hparams: Namespace):
+        """A Pytorch Lightning Module for training/evaluating transformers models on the GLUE benchmark
+
+        Args:
+            hparams (Namespace): Supplied CLI arguments to configure the run.
+        """
         num_labels = glue_tasks_num_labels[hparams.task]
+        hparams.glue_output_mode = "classification" if num_labels > 1 else "regression"
 
         super().__init__(hparams, num_labels, self.mode)
 
     def forward(self, **inputs):
         return self.model(**inputs)
 
     def training_step(self, batch, batch_idx):
-        inputs = {"input_ids": batch[0], "attention_mask": batch[1], "labels": batch[3]}
-
-        if self.config.model_type != "distilbert":
```

Comment:
How do we know it's safe to remove this?

Reply:
Its a little sneaky. I wrangled the correct keys/values in the `prepare_data` function.